dask/dataframe/tests/test_dataframe.py

```diff
@@ -3227,6 +3227,24 @@ def test_apply_warns_with_invalid_meta():
         ddf.apply(func, axis=1, meta=int)
 
 
+@pytest.mark.skipif(not PANDAS_GT_210, reason="Not available before")
+def test_dataframe_map():
+    df = pd.DataFrame({"x": [1, 2, 3, 4], "y": [10, 20, 30, 40]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    assert_eq(ddf.map(lambda x: x + 1), df.map(lambda x: x + 1))
+    assert_eq(ddf.map(lambda x: (x, x)), df.map(lambda x: (x, x)))
+
+
+@pytest.mark.skipif(PANDAS_GT_210, reason="Available at 2.1")
+def test_dataframe_map_raises():
+    df = pd.DataFrame({"x": [1, 2, 3, 4], "y": [10, 20, 30, 40]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    with pytest.raises(
+        NotImplementedError, match="Base package doesn't support 'map'."
+    ):
```

Comment:
This is coming from when we try to get the `pd.DataFrame.map` docstring in the `@derived_from(pd.DataFrame)` decorator. It's nice that we get a `NotImplementedError` error, but something along the lines of (I've not tested / linted the snippet below)

```python
if not PANDAS_GT_210:
    raise NotImplementedError(
        "DataFrame.map requires pandas>=2.1.0, but pandas=X.Y.Z is installed."
    )
```

would, I think, be a bit more clear for users and devs

Reply:
Yeah I had this initially :)

Changed, and removed the derived_from.